### PR TITLE
teststubtest: assert on output without color

### PR DESCRIPTION
**Pull request by ikonst**
These tests have been failing when running in some terminals:
```
FAILED mypy/test/teststubtest.py::StubtestMiscUnit::test_allowlist - AssertionError: assert '\x1b[1m\x1b[...x1b(B\x1b[m\n' == 'Success: no ...in 1 module\n'
FAILED mypy/test/teststubtest.py::StubtestMiscUnit::test_config_file - AssertionError: assert '\x1b[1m\x1b[...x1b(B\x1b[m\n' == 'Success: no ...in 1 module\n'
FAILED mypy/test/teststubtest.py::StubtestMiscUnit::test_ignore_flags - AssertionError: assert '\x1b[1m\x1b[...x1b(B\x1b[m\n' == 'Success: no ...in 1 module\n'
```

### CODE REVIEW

**Comment by ikonst** [comment]
👋 @hauntsaninja

File: mypy/test/teststubtest.py [hunk 1]
```diff
@@ -1866,7 +1865,7 @@ def test_output(self) -> None:
             f"Runtime: in file {TEST_MODULE_NAME}.py:1\ndef (num, text)\n\n"
             "Found 1 error (checked 1 module)\n"
         )
-        assert remove_color_code(output) == expected
```
Comment (ikonst) [inline mypy/test/teststubtest.py:1869]: consolidated in `run_stubtest`

**Review by hauntsaninja — APPROVED** [review]
Looks good! The test failures are probably caused by the new release of typing_extensions and we'd need a typeshed update or something to fix

**Comment by JelleZijlstra** [comment]
I had to deal with the same thing: https://github.com/quora/pyanalyze/commit/a3f8115d14b7f56afe82512c04de399c60d2a30a#diff-6d6be62a11d12e41348b409cfaddd9ea8161e2f78804ae5a7820b6c7197bb8bc. Probably not fixable in typeshed.

**Comment by AlexWaygood** [comment]
> I had to deal with the same thing: [quora/pyanalyze@a3f8115#diff-6d6be62a11d12e41348b409cfaddd9ea8161e2f78804ae5a7820b6c7197bb8bc](https://github.com/quora/pyanalyze/commit/a3f8115d14b7f56afe82512c04de399c60d2a30a#diff-6d6be62a11d12e41348b409cfaddd9ea8161e2f78804ae5a7820b6c7197bb8bc). Probably not fixable in typeshed.

#15294 is the fix.

**Comment by ikonst** [comment]
@hauntsaninja merge?

**Pull request merged**